lib/omt.ejs

```diff
@@ -13,80 +13,61 @@
 
-    if (!registry[name]) {
+  const singleRequire = (uri, parentUri) => {
+    let origURI = uri;
+    uri = new URL(uri + ".js", parentUri).href;
+    return registry[uri] || (
       <% if (useEval) { %>
-        const text = await fetch(url).then(resp => resp.text());
-        eval(text);
+        fetch(uri)
+          .then(resp => resp.text())
+          .then(code => {
+            self.nextDefineUri = uri;
+            eval(code);
+          })
       <% } else { %>
-        if ("document" in self) {
-          await new Promise(resolve => {
+        new Promise(resolve => {
+          if ("document" in self) {
             const script = document.createElement("script");
-            script.src = url;
-            document.head.appendChild(script);
+            script.src = uri;
             script.onload = resolve;
-          });
-        } else {
-          importScripts(url);
-        }
+            document.head.appendChild(script);
+          } else {
+            self.nextDefineUri = uri;
+            importScripts(uri);
+            resolve();
+          }
+        })
       <% } %>
-    }
-    if (!registry[name]) {
-      throw new Error(`Module ${name} didn’t register its module`);
-    }
-    return registry[name];
-  };
-
-  const require = (names, resolve) => {
-    Promise.all(names.map(singleRequire))
-      .then(modules => resolve(modules.length === 1 ? modules[0] : modules));
-  };
-
-  const registry = {
-    require: Promise.resolve(require)
+      .then(() => {
+        let promise = registry[uri];
+        if (!promise) {
+          throw new Error(`Module ${uri} didn’t register its module`);
+        }
+        return promise;
+      })
+    );
   };
 
-  self.<%- amdFunctionName %> = (moduleName, depsNames, factory) => {
-    if (registry[moduleName]) {
+  self.<%- amdFunctionName %> = (depsNames, factory) => {
+    const uri = self.nextDefineUri || ("document" in self ? document.currentScript.src : "") || location.href;
+    if (registry[uri]) {
       // Module is already loading or loaded.
       return;
     }
-    registry[moduleName] = Promise.resolve().then(() => {
-      let exports = {};
-      const module = {
-        uri: location.origin + moduleName.slice(1)
-      };
-      return Promise.all(
-        depsNames.map(depName => {
-          switch(depName) {
-            case "exports":
-              return exports;
-            case "module":
-              return module;
-            default:
-              return singleRequire(depName);
-          }
-        })
-      ).then(deps => {
-        const facValue = factory(...deps);
-        if (!exports.default) {
-          exports.default = facValue;
-        }
-        return exports;
-      });
+    let exports = {};
+    const require = depUri => singleRequire(depUri, uri);
+    const specialDeps = {
+      module: { uri },
+      exports,
+      require
+    };
+    registry[uri] = Promise.resolve().then(() => Promise.all(depsNames.map(
```

Comment:
Comment, so we don't delete this in future 😄

Reply:
Done.